tango/local_workspace.py

```diff
@@ -275,9 +275,38 @@ def work_dir(self, step: Step) -> Path:
         result.mkdir(parents=True, exist_ok=True)
         return result
 
-    @staticmethod
-    def dir_is_empty(dir: Path):
-        return not any(True for _ in dir.iterdir())
+    @classmethod
+    def guess_step_dir_state(cls, dir: Path) -> Optional[StepState]:
+        """
+        Returns the status of a given step dir, to the best of our knowledge.
+
+        :param dir: the step dir to example
+        :return: our best guess at the state of the step, or ``None`` if we don't know
+        """
+
+        # If the directory doesn't exist, the step is incomplete.
+        if not dir.exists():
+            return StepState.INCOMPLETE
+
+        # If the lock file exists and is locked, the step is running.
+        lock_file = dir / "lock"
+        if lock_file.exists():
+            lock = FileLock(lock_file)
+            try:
+                lock.acquire(0)
+                lock.release()
+            except TimeoutError:
+                return StepState.RUNNING
+
+        # If the directory is empty except for the work dir, the step is incomplete.
```

Comment:
Shouldn't we ignore the lock file in addition to the work directory? I believe the lock file will probably exist for any incomplete step, since the lock file is created in the `step_starting()` method. As is, this check will fail to correctly identify a step as `INCOMPLETE` whenever the lock file exists.

Reply:
I think you're right. 81d9276